cpp/include/cudf/reduction/detail/reduction.cuh

```diff
@@ -67,7 +67,16 @@ std::unique_ptr<scalar> reduce(InputIterator d_in,
 {
   auto const binary_op     = cudf::detail::cast_functor<OutputType>(op.get_binary_op());
   auto const initial_value = init.value_or(op.template get_identity<OutputType>());
-  auto dev_result          = rmm::device_scalar<OutputType>{initial_value, stream, mr};
+  auto host_scalar =
+    cudf::detail::make_pinned_vector_async<OutputType>(1, stream);  // as host pinned memory
+  CUDF_CUDA_TRY(cudaMemcpyAsync(
```

Comment:
Can we use any of the `cudf::detail::cuda_memcpy_async` overloads here and everywhere we are directly calling `cudaMemcpyAsync`?